haystack/pipelines/base.py

```diff
@@ -578,6 +578,132 @@ def run(  # type: ignore
         self.send_pipeline_event_if_needed(is_indexing=file_paths is not None)
         return node_output
 
+    # async version of the above `run()` method
+    async def run_async(  # type: ignore
```

Comment:
I think for now it would make more sense to have this `run_async` in `RayPipeline`, because on regular Pipelines it will simply fail due to the lack of `_run_node_async`? Is there any concern moving it there?

Reply:
No concern, that is a great idea.
I originally put `run_async` into `Pipeline` instead of `RayPipeline`, so it can be right next to the original, sync version (eg the `run` method), but you are right, it would make more sense for it to be in the `RayPipeline`.
I am moving that now and sending in a commit with that (which also makes that `type: ignore` unnecessary - see your other comment above`)